options/internal/generic/search.cpp

```diff
@@ -0,0 +1,98 @@
+#include <mlibc/search.hpp>
+#include <frg/string.hpp>
+#include <stdlib.h>
+#include <stdint.h>
+#include <errno.h>
+
+struct _ENTRY {
+	ENTRY entry;
+	bool used;
+};
+
+namespace mlibc {
+
+int hcreate_r(size_t num_entries, struct hsearch_data *htab) {
+	if(!htab) {
+		errno = EINVAL;
+		return 0;
+	}
+
+	htab->table = static_cast<_ENTRY*>(calloc(1, num_entries * sizeof(_ENTRY)));
```

Comment:
Doing `calloc(num_entries, sizeof(_ENTRY))` (to prevent multiplication overflows) or using `getAllocator().allocate` might be better here.